### PR TITLE
Add a _spendAllowance function to ERC20 & ERC777

### PR DESCRIPTION
**Pull request by Amxx**
- Add a `_spendAllowance` in ERC20
- Add a `_spendAllowance` in ERC777
- ~~Rename some arguments in ERC777 for clarity (similar to #3167)~~
- ERC20Burnable: add missing behavior from in #3085

#### PR Checklist

- [x] Tests
- [ ] Documentation
- [x] Changelog entry

### CODE REVIEW

**Comment by frangio** [comment]
Are we adding an internal `_spendAllowance` for use in both `transferFrom` and `burnFrom`?

File: contracts/token/ERC777/ERC777.sol [hunk 1]
```diff
@@ -331,47 +330,47 @@ contract ERC777 is Context, IERC777, IERC20 {
...
+     * - `to` cannot be the zero address.
+     * - if `to` is a contract, it must implement the {IERC777Recipient}
      * interface.
      */
     function _mint(
-        address account,
+        address to,
         uint256 amount,
-        bytes memory userData,
+        bytes memory data,
```
Comment (frangio) [inline contracts/token/ERC777/ERC777.sol:340]: Given that there is `operatorData` I think it makes sense for this to be `userData`.

**Comment by Amxx** [comment]
> Sorry but the ERC777 naming changes are too much to include in an unrelated PR... Can you please split that out?

done

**Review by frangio — APPROVED** [review]
Thank you!

**Pull request merged**